ROOTFS/etc/waggle/sanity/fatal/check_hanwha_camera.test

```diff
@@ -0,0 +1,119 @@
+    else:
+        print(f'{title}: {msg} FAIL')
+
+
+def read_camera_manifest(topic_camera='cameras', path='/etc/waggle/node_manifest.json'):
+    if not os.path.exists(path):
+        raise FileNotFoundError(f'manifest does not exist: {path}')
+
+    with open(path, 'r') as file:
+        try:
+            manifest = json.loads(file.read())
+        except:
+            raise IOError(f'Failed to load the manifest: {path}')
+        if topic_camera not in manifest:
+            raise EOFError(f'no cameras listed in the manifest, skipping test')
+        else:
+            for camera in manifest[topic_camera]:
+                if camera['present'] == True:
+                    yield camera
+
+
+def http_request(url, user='waggle', pw='a2kWe89wak3na7@', timeout=5):
+    passman = urllib.request.HTTPPasswordMgrWithDefaultRealm()
+    passman.add_password(None, url, user, pw)
+    authhandler = urllib.request.HTTPDigestAuthHandler(passman)
+    opener = urllib.request.build_opener(authhandler)
+    urllib.request.install_opener(opener)
+    headers = {'Accept': 'application/json'}
+    req = urllib.request.Request(url, headers=headers)
+    try:
+        with urllib.request.urlopen(req, timeout=timeout) as response:
+            return response.code, response.read()
+    except urllib.error.URLError:
+        return 404, b''
+
+
+def check(camera_ip_domain='10.31.81.'):
+    try:
+        cameras = list(read_camera_manifest())
+    except FileNotFoundError as ex:  # failed to locate the manifest
+        print_log(str(ex), is_pass=False)
+        return (1 << NO_NODE_MANIFEST)
+    except IOError as ex:  # failed to read the manifest
+        print_log(str(ex), is_pass=False)
+        return (1 << READ_ERROR_MANIFEST)
+    except EOFError as ex:  # if no cameras provided, no need to test
+        print_log(str(ex), is_pass=True)
+        return 0x00
+
+    error_code = 0x00
+    for n in range(10, 20, 1):
+        # get device info
+        return_code, body_json = http_request(f'http://{camera_ip_domain}{n}/stw-cgi/system.cgi?msubmenu=deviceinfo&action=view')
+        if return_code != 200:
+            # only properly provisioned Hanwha camera should be able to answer to this call
+            # NOTE: unauthorized error 401 is not captured in the http response
+            #       it is encoded inside the body by Hanwha SUN API
+            #if return_code == 401:
+            #    print_log(f'access to {camera_ip_domain}{n} denied;unauthorized', is_pass=False)
+            continue
+        device_info = json.loads(body_json)
+
+        return_code, body_json = http_request(f'http://{camera_ip_domain}{n}/stw-cgi/video.cgi?msubmenu=snapshot&action=view')
+        if return_code != 200:
+            print_log(f'failed to get a snapshot from {camera_ip_domain}{n}', is_pass=False)
+            error_code = error_code | (1 << FAILED_TO_GET_SNAPSHOT)
+
+        # check if the device model is listed in manifest
+        camera_to_delete = []
```

Comment:
I think Joe already mentioned this but this code looks like it's basically doing a "find first camera with model name" search. You could probably just use None instead of [].

Reply:
That would simplify some of your "for c in camera_to_delete:..." steps to just cameras.remove(camera_to_delete).

Reply:
Changed as suggested!